MuvrKit/MKWatchConnectivity.swift

```diff
@@ -242,58 +328,55 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
             let from = props.accelerometerStart ?? session.start
             let to = props.end ?? NSDate()
             
-            guard let sensorData = getSamples(from: from, to: to, demo: session.demo) else {
+            guard let (fileUrl, end) = encodeSamples(from: from, to: to, demo: session.demo) else {
                 NSLog("No sensor data in \(from) - \(to)")
                 return
             }
-
+            
             // update the number of recorded samples
-            let readFromDate = NSDate(timeIntervalSince1970: sensorData.end)
-            let updatedProps = props.with(accelerometerEnd: readFromDate)
-            self.sessions[session] = updatedProps
+            let updatedProps = props.with(accelerometerEnd: end)
```

Comment:
I'd prefer use the ``update`` with a function argument as props may be out of date (was read before encoding the data).